.NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishDatePeriodParserConfiguration.cs

```diff
@@ -209,7 +207,7 @@ public bool IsLastCardinal(string text)
         public bool IsMonthOnly(string text)
         {
             var trimedText = text.Trim().ToLowerInvariant();
-            return trimedText.EndsWith("month") || trimedText.EndsWith(string.Join(" ", "month", AfterNextSuffix));
+            return trimedText.EndsWith("month") || Regex.IsMatch(text, @"\bmonth\b") && AfterNextSuffixRegex.IsMatch(trimedText);
```

Comment:
We should avoid using Regex.IsMatch if possible. Especially if there are too many uses of it.
Can you try to refactor? Please reach out if not sure.
Thanks.